addons/project_hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,14 @@
+from odoo import models
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    def _compute_analytic_distribution(self):
+        project_id = self.env.context.get('project_id')
+        if not project_id:
+            super()._compute_analytic_distribution()
+        else:
+            analytic_account = self.env['project.project'].browse(project_id).analytic_account_id
+            for expense in self:
+                expense.analytic_distribution = expense.analytic_distribution or {analytic_account.id: 100}
```

Comment:
```suggestion
        project_id = self.env.context.get('project_id')
        if project_id:
            project_analytic_distribution = {self.env['project.project'].browse(project_id).analytic_account_id.id: 100}
            for expense in self:
                expense.analytic_distribution = expense.analytic_distribution or project_analytic_distribution
        else:
            super()._compute_analytic_distribution()
```
Nitpick, I often do the same but it's easier to understant if you avoid the negation when possible
+ Let's avoid redefining the same dict at every loop, doesn't cost much but it's still energy saved
:)